toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -87,3 +98,29 @@ func compressWithZstd(partitionRawFilepath string) (partitionFilepath string, er
 
 	return partitionRawFilepath + ".zst", nil
 }
+
+// Write partition metadata as JSON to a file.
+func writePartitionMetadataJson(diskPartition []diskutils.PartitionInfo, outDir string, name string) (err error) {
```

Comment:
The toolkit has a [jsonutils.WriteJSONFile](https://github.com/microsoft/azurelinux/blob/2.0/toolkit/tools/internal/jsonutils/jsonutils.go#L77C6-L77C19) function.